pkg/testcase/daemonset.go

```diff
@@ -17,17 +21,77 @@ func TestDaemonset(deployWorkload bool) {
 		Expect(err).NotTo(HaveOccurred(),
 			"Daemonset manifest not deployed")
 	}
-	nodes, _ := shared.ParseNodes(false)
 	pods, _ := shared.ParsePods(false)
 
-	var allNodes []shared.Node
-	for _, node := range nodes {
-		if !(node.Roles == "control-plane,master" || node.Roles == "etcd") {
-			allNodes = append(allNodes, node)
+	cmd := fmt.Sprintf(`
+		kubectl get pods -n test-daemonset -o wide --kubeconfig="%s" \
+		| grep -A10 NODE | awk 'NR>1 {print $7}'
+		`,
+		shared.KubeConfigFile,
+	)
+	nodeNames, err := shared.RunCommandHost(cmd)
+	if err != nil {
+		return
+	}
+
+	var nodes []string
+	n := strings.Split(nodeNames, "\n")
+	for _, nodeName := range n {
+		if nodeName != "" {
+			nodes = append(nodes, nodeName)
 		}
 	}
+
+	cmd = fmt.Sprintf(`
+		kubectl get nodes -o custom-columns=NAME:.metadata.name,TAINTS:.spec.taints \
+		--kubeconfig="%s" | grep '<none>'
+		`,
+		shared.KubeConfigFile,
+	)
+	taints, err := shared.RunCommandHost(cmd)
+	if err != nil {
+		return
+	}
+	Expect(taints).To(ContainSubstring("<none>"))
+	Expect(validateNodes(strings.TrimSpace(nodeNames), strings.TrimSpace(taints))).To(BeTrue())
```

Comment:
params are passed in a different order than it is received. functionality would not be affected irrespective i guess

Reply:
yeah, but better correct,  doing now.
tks